client/v3/watch_test.go

```diff
@@ -53,3 +56,59 @@ func TestEvent(t *testing.T) {
 		}
 	}
 }
+
+// TestStreamKeyFromCtx tests the streamKeyFromCtx function to ensure it correctly
+// formats metadata as a map[string][]string when extracting metadata from the context.
+//
+// The fmt package in Go guarantees that maps are printed in a consistent order,
+// sorted by the keys. This test verifies that the streamKeyFromCtx function
+// produces the expected formatted string representation of metadata maps when called with
+// various context scenarios.
+func TestStreamKeyFromCtx(t *testing.T) {
+	tests := []struct {
+		name     string
+		ctx      context.Context
+		expected string
+	}{
+		{
+			name: "multiple keys",
+			ctx: metadata.NewOutgoingContext(context.Background(), metadata.MD{
+				"key1": []string{"value1"},
+				"key2": []string{"value2a", "value2b"},
+			}),
+			expected: "map[key1:[value1] key2:[value2a value2b]]",
+		},
+		{
+			name:     "no keys",
+			ctx:      metadata.NewOutgoingContext(context.Background(), metadata.MD{}),
+			expected: "map[]",
+		},
+		{
+			name: "only one key",
+			ctx: metadata.NewOutgoingContext(context.Background(), metadata.MD{
+				"key1": []string{"value1", "value1a"},
+			}),
+			expected: "map[key1:[value1 value1a]]",
+		},
+		{
+			name:     "no metadata",
+			ctx:      context.Background(),
+			expected: "",
+		},
+	}
+
+	// Iterate through each test case and verify the function's output matches
+	// the expected result.
+	for _, tt := range tests {
+		// Capture range variable to ensure each closure gets its own copy of tt.
+		tt := tt
```

Comment:
FYI. golang 1.22 already resolved this issue (variables declared by a “for” loop overwritten by next iteration), it isn't necessary anymore starting from 1.22. But feel free to keep it as it's.